lib/rails_config_validator/validator.rb

```diff
@@ -0,0 +1,83 @@
+require 'kwalify'
+require 'yaml'
+require 'rails_config_validator/errors'
+
+module RailsConfigValidator
+  class Validator
+    def initialize(config_path, env, schema_path: nil)
+      @config_path = config_path
+      @schema_path = schema_path || config_to_schema_path(config_path)
+      @env = env
+    end
+
+    def valid?
+      meta_validate.empty? && validate.empty?
+    rescue
+      # ignored
+      false
+    end
+
+    def valid!
+      mv = meta_validate
+      fail InvalidSchemaError,
+           "Incorrect schema #{schema_path}; errors: #{mv}" unless mv.empty?
+      v = validate
+      fail InvalidConfigurationError,
+           "Incorrect config #{config_path}; errors: #{v}" unless v.empty?
+    end
+
+    def meta_validate
```

Comment:
`meta_validate` and `validate` both throw exceptions so I'd put a bang in the method name like in the `valid!` method.
Do they have to be `public`?

On failure they return an array of errors and you can't infer it based on the method name. Maybe it makes sense to have an `errors` method on the Validator instance instead?